src/relay.js

```diff
@@ -288,12 +292,12 @@ export function sequelizeConnection({
         // In case of `OVER()` is not available, we need to get the full count from a second query.
         const options = await Promise.resolve(before({
           where: argsToWhere(args)
-        }));
+        }, args, context, info));
 
         if (target.count) {
-          fullCount = await target.count(source, options);
+          fullCount = await target.count(options);
         } else {
-          fullCount = await target.manyFromSource.count(source, options);
+          fullCount = await target.manyFromSource.count(options);
```

Comment:
Not sure about this one actually. Target can both be an association (in which case passing the source instance as the first arg is correct (https://github.com/sequelize/sequelize/blob/master/lib/associations/has-many.js#L342)), but it can also be a model - In which case we shouldn't pass the source.

Probably missing a count test for the latter case